.github/workflows/mpi-tests.yml

```diff
@@ -26,11 +26,11 @@ jobs:
             python-version: "3.10"
             mpi: intelmpi
             jax-version: "latest"
-          # test older JAX and Python versions
+          # test older Python versions
           - os: ubuntu-latest
-            python-version: "3.6"
+            python-version: "3.7"
             mpi: openmpi
-            jax-version: "0.2.17"  # last version with Python 3.6 support
+            jax-version: "latest"
```

Comment:
Do we still have a tester running the oldest version of jax/jaxlib?

Reply:
Otherwise I'll make a PR to run tests on oldest dependencies

Reply:
I'll figure out what the new JAX requirement is and add a test.

Reply:
Turns out `Primitive.get_bind_params` was introduced in JAX 0.2.27 (released 2 weeks ago). That seems... restrictive?

Reply:
We can either have experimental check jax version and it only works on very recent versions,
Or drop support for older versions altogether.

While I dislike dropping support, current release has been relatively stable and fixed several bugs so I wouldn't be against

Reply:
If any bug fix will be really needed we can always backport them to a 0.3.x branch

Reply:
> We can either have experimental check jax version and it only works on very recent versions

I very much prefer this, and we already have functions to check versions so it should be a simple implementation. We can always drop old versions later on. Closing this PR for now, feel free to re-open if you disagree.

Reply:
I don't feel strongly about it.